Makefile

```diff
@@ -76,7 +76,7 @@ FUNCTIONAL_TEST_XDC_ROOT      := ./tests/xdc
 FUNCTIONAL_TEST_NDC_ROOT      := ./tests/ndc
 DB_INTEGRATION_TEST_ROOT      := ./common/persistence/tests
 DB_TOOL_INTEGRATION_TEST_ROOT := ./tools/tests
-INTEGRATION_TEST_DIRS := $(DB_INTEGRATION_TEST_ROOT) $(DB_TOOL_INTEGRATION_TEST_ROOT)
+INTEGRATION_TEST_DIRS := $(DB_INTEGRATION_TEST_ROOT) $(DB_TOOL_INTEGRATION_TEST_ROOT) ./temporaltest
```

Comment:
🤔 Right now `temporaltest` is being run with both unit and integration test targets. In CI it consistently [passes in unit tests](https://buildkite.com/temporal/temporal-public/builds/9014#0187ba4b-52cc-4298-92f7-f6aef238cb97/153-380), but [fails in integration tests](https://buildkite.com/temporal/temporal-public/builds/9014#0187ba4b-52cd-4c12-8de1-6fce84784ece/138-278).

I can't reproduce this failure locally either with `go test ./temporaltest` or `make integration-test-coverage` (the latter after removing `DB_INTEGRATION_TEST_ROOT` and `DB_TOOL_INTEGRATION_TEST_ROOT` on this line since I don't have DBs running).

Any ideas?

Reply:
Update: I'm not able to reproduce the integration test failures locally after starting up dependencies using `make start-dependencies`:

```
$ make integration-test
Run integration tests...
ok  	go.temporal.io/server/common/persistence/tests	223.242s
ok  	go.temporal.io/server/tools/tests	23.265s
ok  	go.temporal.io/server/temporaltest	13.404s
```

Reply:
Actually I've gotten a failure through `make unit-test` as well, though it's a data race:

<details>

```
$ make unit-test-coverage
Run unit tests with coverage...
==================
WARNING: DATA RACE
Read at 0x00c009498840 by goroutine 14464:
  runtime.mapaccess1_faststr()
      /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/map_faststr.go:13 +0x41c
  github.com/temporalio/tchannel-go/json.Register.func1()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/json/handler.go:100 +0x78
  github.com/temporalio/tchannel-go.HandlerFunc.Handle()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/handlers.go:49 +0x48
  github.com/temporalio/tchannel-go.(*handlerMap).Handle()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/handlers.go:123 +0x26c
  github.com/temporalio/tchannel-go.channelHandler.Handle()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/handlers.go:131 +0x9c
  github.com/temporalio/tchannel-go.(*Connection).dispatchInbound()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/inbound.go:203 +0x694
  github.com/temporalio/tchannel-go.(*Connection).handleCallReq.func3()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/inbound.go:125 +0x64

Previous write at 0x00c009498840 by goroutine 14376:
  runtime.mapaccess2_faststr()
      /opt/homebrew/Cellar/go/1.19.5/libexec/src/runtime/map_faststr.go:108 +0x43c
  github.com/temporalio/tchannel-go/json.Register()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/json/handler.go:119 +0x238
  github.com/temporalio/ringpop-go/swim.(*Node).registerHandlers()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/ringpop-go@v0.0.0-20220818230611-30bf23b490b2/swim/handlers.go:81 +0xa8c
  github.com/temporalio/ringpop-go/swim.NewNode()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/ringpop-go@v0.0.0-20220818230611-30bf23b490b2/swim/node.go:278 +0x99c
  github.com/temporalio/ringpop-go.(*Ringpop).init()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/ringpop-go@v0.0.0-20220818230611-30bf23b490b2/ringpop.go:195 +0x420
  github.com/temporalio/ringpop-go.(*Ringpop).Bootstrap()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/ringpop-go@v0.0.0-20220818230611-30bf23b490b2/ringpop.go:419 +0x3c
  go.temporal.io/server/common/membership/ringpop.(*service).bootstrap.func1()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/common/membership/ringpop/ringpop.go:108 +0x1a8
  go.temporal.io/server/common/backoff.ThrottleRetry.func1()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/common/backoff/retry.go:175 +0x30
  go.temporal.io/server/common/backoff.ThrottleRetryContext()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/common/backoff/retry.go:199 +0x158
  go.temporal.io/server/common/backoff.ThrottleRetry()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/common/backoff/retry.go:176 +0x78
  go.temporal.io/server/common/membership/ringpop.(*service).bootstrap()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/common/membership/ringpop/ringpop.go:114 +0x128
  go.temporal.io/server/common/membership/ringpop.(*service).start()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/common/membership/ringpop/ringpop.go:84 +0x50
  go.temporal.io/server/common/membership/ringpop.(*monitor).Start()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/common/membership/ringpop/monitor.go:139 +0x3c0
  go.temporal.io/server/common/membership/ringpop.(*monitor).Start-fm()
      <autogenerated>:1 +0x34
  go.uber.org/fx/internal/lifecycle.Wrap[...].func1()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/internal/lifecycle/lifecycle.go:80 +0x30
  go.uber.org/fx/internal/lifecycle.(*Lifecycle).runStartHook()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/internal/lifecycle/lifecycle.go:256 +0x218
  go.uber.org/fx/internal/lifecycle.(*Lifecycle).Start()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/internal/lifecycle/lifecycle.go:216 +0x408
  go.uber.org/fx.(*App).start.func1()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/app.go:678 +0x64
  go.uber.org/fx.(*App).withRollback()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/app.go:660 +0x48
  go.uber.org/fx.(*App).start()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/app.go:677 +0x50
  go.uber.org/fx.(*App).start-fm()
      <autogenerated>:1 +0x44
  go.uber.org/fx.withTimeout.func1()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/app.go:781 +0xac

Goroutine 14464 (running) created at:
  github.com/temporalio/tchannel-go.(*Connection).handleCallReq()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/inbound.go:125 +0x1598
  github.com/temporalio/tchannel-go.(*Connection).handleFrameNoRelay()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/connection.go:709 +0xc4
  github.com/temporalio/tchannel-go.(*Connection).readFrames()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/connection.go:676 +0x14c
  github.com/temporalio/tchannel-go.(*Channel).newConnection.func1()
      /Users/jacoblegrone/go/pkg/mod/github.com/temporalio/tchannel-go@v1.22.1-0.20220818200552-1be8d8cffa5b/connection.go:374 +0x40

Goroutine 14376 (running) created at:
  go.uber.org/fx.withTimeout()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/app.go:769 +0x120
  go.uber.org/fx.(*App).Start()
      /Users/jacoblegrone/go/pkg/mod/go.uber.org/fx@v1.19.1/app.go:646 +0x280
  go.temporal.io/server/temporal.(*ServerImpl).startServices.func1()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/temporal/server_impl.go:158 +0x58
  go.temporal.io/server/temporal.(*ServerImpl).startServices.func2()
      /Users/jacoblegrone/Development/github.com/temporalio/temporal/temporal/server_impl.go:163 +0x44
==================
--- FAIL: TestBaseServerOptions (0.18s)
    testing.go:1319: race detected during execution of test
FAIL
coverage: 77.1% of statements
FAIL	go.temporal.io/server/temporaltest	9.363s
FAIL
make: *** [unit-test-coverage] Error 1
```

</details>

Perhaps we should quarantine `temporaltest` in an internal package for now as well until we know it can be used reliably?